packages/medusa/src/loaders/index.ts

```diff
@@ -149,12 +151,12 @@ async function loadMedusaV2({
   await createDefaultsWorkflow(container).run()
 
   const shutdown = async () => {
-    await promiseAll([
-      container.dispose(),
-      pgConnection?.context?.destroy(),
-      expressShutdown(),
-      medusaAppOnApplicationShutdown()
-    ])
+    await medusaAppOnApplicationShutdown(),
+      await promiseAll([
+        container.dispose(),
+        pgConnection?.context?.destroy(),
+        expressShutdown(),
```

Comment:
Q: should we move that one before the medusa app shutdown to ensure no more requests can be handled?

Reply:
Yeah I think that makes sense, cc @edast will you take over the PR?